Tasks/AzureRmWebAppDeployment/AzureRMUtil.ts

```diff
@@ -0,0 +1,293 @@
+import Q = require('q');
+
+var AuthenticationContext = adal.AuthenticationContext;
+
+function getAuthorizationToken ( SPN ) : Q.Promise <string> {
+
+	var deferred = Q.defer<string>();
+	var authorityUrl = 'https://login.windows.net/' + SPN.tenantID; 
+	var resource = 'https://management.azure.com/';
+
+	var context = new AuthenticationContext (authorityUrl);
+	context.acquireTokenWithClientCredentials (resource, SPN.servicePrincipalClientID, SPN.servicePrincipalKey, function ( error, tokenResponse) {
+		if ( error ) 
+			deferred.reject ( error );
+		else {
+			deferred.resolve ( tokenResponse.accessToken );
+		}
+	});
+
+	return deferred.promise;
+}
+
+exports.getAzureRMWebAppDetails = function ( SPN, webAppName : string ) : Q.Promise<any> {
+
+	var deferred = Q.defer<any> ();
+
+	getAuthorizationToken(SPN)
+	.then(function (accessToken : string) {
+		
+ 		var requestOptions = {
+ 		 	url : 'https://management.azure.com/subscriptions/' + SPN.subscriptionId + '/resources?$filter=resourceType eq \'Microsoft.Web%2FSites\'&api-version=2016-02-01',
+ 		 	auth : {
+ 		 		bearer : accessToken
+ 		 	}
+ 		 };
+
+ 		request (requestOptions, function (error, response, body) {
+ 		 	if ( error ) 
+ 		 		deferred.reject (error);
+ 		 	else {
+ 		 		if ( response.statusCode === 200 ) {
+ 		 			var obj = JSON.parse (body);
+ 		 			for ( var i = 0; i < Object.keys (obj.value).length; i++ ) {
+ 		 				if ( obj.value[i].name == webAppName ) {
+ 		 					deferred.resolve (obj.value[i]);
+						}
+					}
+ 		 			deferred.reject ('Error Occurred : No such web app');
+ 		 		}
+ 		 		else deferred.reject ('Error Occurred : [Staus Code : '+response.statusCode+']');
+
+ 		 	}
+ 		});
+ 		
+	}, function (error) {
+		taskLib.error (error);
+	});
+
+	return deferred.promise;
+
+};
+
+function getPublishProfileUtil ( SPN, webAppName: string, resourceGroupName: string, publishMethod: string, deployToSlotFlag: boolean, slotName: string ): Q.Promise<any> {
+
+    var deferred = Q.defer();
+    var insertInUrl = deployToSlotFlag ? "/slots/" + slotName : "";
+    getAuthorizationToken(SPN)
+        .then(function (accessToken) {
+        var requestOptions = {
+            url: 'https://management.azure.com/subscriptions/' + SPN.subscriptionId + '/resourceGroups/' + resourceGroupName + '/providers/Microsoft.Web/sites/' + webAppName + insertInUrl + '/publishxml?api-version=2015-08-01',
+            auth: {
+                bearer: accessToken
+            },
+            method: 'POST'
+        };
+        request(requestOptions, function (error, response, body) {
+            if (error)
+                deferred.reject(error);
+            else {
+                if (response.statusCode === 200) {
+                    parseString(body, function (error, result) {
+                        for (var i in result.publishData.publishProfile) {
+                            if (result.publishData.publishProfile[i].$.publishMethod === publishMethod)
```

Comment:
By default , publishMethod having value "Select FTP or Web Deploy" causing task to fail, Provide some default value.